pkg/awsutils/awsutils.go

```diff
@@ -602,90 +608,96 @@ func (cache *EC2InstanceMetadataCache) getENIMetadata(eniMAC string) (ENIMetadat
 		deviceNum = 0
 	}
 
-	log.Debugf("Found ENI: %s, MAC %s, device %d", eniID, eniMAC, deviceNum)
-
-	// Get IPv4 and IPv6 addresses assigned to interface
-	cidr, err := cache.imds.GetSubnetIPv4CIDRBlock(ctx, eniMAC)
-	if err != nil {
-		awsAPIErrInc("GetSubnetIPv4CIDRBlock", err)
-		return ENIMetadata{}, err
-	}
-
-	imdsIPv4s, err := cache.imds.GetLocalIPv4s(ctx, eniMAC)
-	if err != nil {
-		awsAPIErrInc("GetLocalIPv4s", err)
-		return ENIMetadata{}, err
-	}
-
-	ec2ip4s := make([]*ec2.NetworkInterfacePrivateIpAddress, len(imdsIPv4s))
-	for i, ip4 := range imdsIPv4s {
-		ec2ip4s[i] = &ec2.NetworkInterfacePrivateIpAddress{
-			Primary:          aws.Bool(i == 0),
-			PrivateIpAddress: aws.String(ip4.String()),
-		}
-	}
+	log.Debugf("Found ENI: %s, MAC %s, device %d, network card %d", eniID, eniMAC, deviceNum, networkCard)
 
+	var subnetV4Cidr string
+	var ec2ip4s []*ec2.NetworkInterfacePrivateIpAddress
 	var ec2ip6s []*ec2.NetworkInterfaceIpv6Address
 	var subnetV6Cidr string
-	if cache.v6Enabled {
-		// For IPv6 ENIs, do not error on missing IPv6 information
-		v6cidr, err := cache.imds.GetSubnetIPv6CIDRBlocks(ctx, eniMAC)
-		if err != nil {
-			awsAPIErrInc("GetSubnetIPv6CIDRBlocks", err)
-		} else {
-			subnetV6Cidr = v6cidr.String()
-		}
+	var ec2ipv4Prefixes []*ec2.Ipv4PrefixSpecification
+	var ec2ipv6Prefixes []*ec2.Ipv6PrefixSpecification
 
-		imdsIPv6s, err := cache.imds.GetIPv6s(ctx, eniMAC)
+	if networkCard == 0 {
```

Comment:
What are we verifying with a `if networkCard == 0` here? 

1. Can this condition ever be false? 
2. Should we add a comment that explains why we are checking this here?

Also, we need to test this change in some variety of instances - Multi Card Instance, regular instance, and ensuring this is not leading to any unexpected behavior.

Reply:
@orsenthil On a multicard instance this condition would be false for non-0 cards. For non-0 cards we are just getting basic eni info and not fetching any ip related info as some of the efa interfaces might not have ip  info

Here https://github.com/aws/amazon-vpc-cni-k8s/blob/master/pkg/awsutils/awsutils.go#L1344 we already assume CNI always manages only network card 0 and rest are just added as unmanaged ENI's so it is safe to make the assumption that we need complete ENIMetadata only for network card 0  

I have tested this on multiple instances including multi-card, single card, different efa interfaces and verified the functionality.

I will add comments in code explaining why we are checking for network card 0